File: java/src/main/java/com/smeltery/service/config/AsyncConfig.java
package com.smeltery.service.config;

import com.smeltery.service.exception.CustomAsyncExceptionHandler;
import java.util.concurrent.Executor;
import java.util.concurrent.ThreadPoolExecutor;
import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.AsyncConfigurer;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

@Configuration
@EnableAsync
public class AsyncConfig implements AsyncConfigurer {

    @Override
    public Executor getAsyncExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();

		executor.setMaxPoolSize(100);
        executor.setCorePoolSize(75);
        executor.setQueueCapacity(75);
        executor.setRejectedExecutionHandler(new ThreadPoolExecutor.CallerRunsPolicy());
        executor.setThreadNamePrefix("Executor-");
        executor.initialize();

		return executor;
    }

    @Override
    public AsyncUncaughtExceptionHandler getAsyncUncaughtExceptionHandler() {
        return new CustomAsyncExceptionHandler();
    }
}

File: kotlin/src/main/kotlin/com/ibm/service/config/AsyncConfig.kt
package com.smeltery.service.config

import com.smeltery.service.exceptions.CustomAsyncExceptionHandler
import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.AsyncConfigurer
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import java.util.concurrent.Executor
import java.util.concurrent.ThreadPoolExecutor

@Configuration
@EnableAsync
class AsyncConfig : AsyncConfigurer {
	override fun getAsyncExecutor(): Executor? {
		val executor = ThreadPoolTaskExecutor()
		executor.maxPoolSize = 100
		executor.corePoolSize = 75
		executor.queueCapacity = 75
		executor.setRejectedExecutionHandler(ThreadPoolExecutor.CallerRunsPolicy())
		executor.setThreadNamePrefix("Executor-")
		executor.initialize()
		return executor
	}

	override fun getAsyncUncaughtExceptionHandler(): AsyncUncaughtExceptionHandler? {
		return CustomAsyncExceptionHandler()
	}
}
